### ISSUE TITLE
🐛 [Nextcloud] Unable to use Collabora Online

### ISSUE THREAD

**Issue by ShonP40** [alexbelgium/hassio-addons#1028]
### Description

I’m trying to setup the CODE server but Nextcloud doesn’t seem to be able to connect to it

### Reproduction steps

```bash
1. Install the CODE server
2. Install Nextcloud Office
3. Go to the Nextcloud Office settings page
4. An error will be displayed about not being able to connect to the CODE server
```

### Addon Logs

```bash
Just showing this over and over again

172.70.238.131 - - [15/Oct/2023:00:33:11 +0300] "GET /apps/richdocumentscode_arm64/proxy.php?req=/hosting/capabilities HTTP/1.1" 404 1716 "-" "Nextcloud Server Crawler"
```

### Architecture

aarch64

### OS

HAos

**Comment by ShonP40** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/13995143/cb2b01f4-3c4c-49d3-b700-6c0d7afb2918)

**Comment by ShonP40** [comment]
Also, I noticed that the CODE app keeps getting uninstalled on every restart.

**Comment by alexbelgium** [comment]
Hi, here is the reason : https://github.com/linuxserver/docker-nextcloud#collaborative-editing

**Comment by ShonP40** [comment]
They added an ARM64 version though

**Comment by ShonP40** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/13995143/4ab404b9-2e3a-4a10-9552-8bf7e360bdb9)

**Comment by alexbelgium** [comment]
OK thanks ; I think you need to do a request on lsio github as they are removing it here I believe : https://github.com/linuxserver/docker-nextcloud/blob/2ecd6cb8d9843ae1d87fe0b7317dc2179d5363fd/root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run#L144

**Comment by ShonP40** [comment]
That script only runs on startup tho
So still doesn’t explain why it’s not working when it is installed (without a restart)

**Comment by ShonP40** [comment]
Pretty sure that it has something to do with the Nginx config

**Comment by alexbelgium** [comment]
Well I tried installing code and it crashed my installation so perhaps lsio is right about incompatibility

**Comment by ShonP40** [comment]
Well, I did notice that my install was freezing sometimes.
Could’ve been related to having that installed

**Comment by alexbelgium** [comment]
I think that's why I had also added this piece of code to disable code at boot, otherwise system can't be recovered except by using portainer. So, the path seems to be an external office container rather than the built in CODE server...

**Comment by ShonP40** [comment]
Also, I know that this is kinda unrelated, but is there a way to increase the PHP limits?
They’re kinda too low

![image](https://github.com/alexbelgium/hassio-addons/assets/13995143/09a19647-b769-4912-a321-03be8554e613)

**Comment by alexbelgium** [comment]
> Also, I know that this is kinda unrelated, but is there a way to increase the PHP limits? They’re kinda too low
> 
> ![image](https://user-images.githubusercontent.com/13995143/275353143-09a19647-b769-4912-a321-03be8554e613.png)
> 
> Would be nice to just have it in the addon configuration

Hi, implemented. I've reimplemented the script to remove CODE on boot as my system wasn't starting otherwise

**Comment by ShonP40** [comment]
Alright, thank you!
I guess we’ll need to wait for the team behind LinuxServer to come up with a fix for CODE.

**Comment by dkahn93** [comment]
Reopening this, because the same thing is happening to me, and I noticed it's been awhile since the last post here. Has there been any fix since? I install Collabora Built-in CODE Server (Arm64) and the logs just repeat:
"GET /apps/richdocumentscode_arm64/proxy.php?status HTTP/1.1" 499 0 "-" "Nextcloud Server Crawler"
Nextcloud Office settings show: 
<img width="863" alt="nextclouddoc" src="https://github.com/user-attachments/assets/75476393-4958-4099-bb12-0da7c2246b6f">
Nextcloud all but crashes. I have to restart the HA add-on and go quickly to disable the CODE server before the CODE server I guess starts back up again, and then it works fine again.

**Comment by ShonP40** [comment]
Well I kind ditched NextCloud completely a while ago and not planning on reinstalling it so I have no idea if it was fixed or not

**Comment by aelg305** [comment]
@dkahn93 did you figure out a solution?

Im looking for either an AddOn to Hass with collabora or... I have the following...havent tested and figure I could inject the code as @alexbelgium has shown with the script?? dont know...just throwing solutions out there:

Collabora - potentially getting it to works: 
    # from https://www.collaboraonline.com/blog/connecting-collabora-online-built-in-code-server-with-nginx/
    # As per documents, need to edit/change or add/rewrite: 
    in: /data/config/nginx/site-confs/default.conf
    from :  
    
    location ~ \.php(?:$|/) {
            # Required for legacy support
            rewrite ^/(?!index|remote|public|cron|core\/ajax\/update|status|ocs\/v[12]|updater\/.+|ocs-provider\/.+|.+\/richdocumentscode\/proxy) /index.php$request_uri;
                                                                                                                                                                                                            
            fastcgi_split_path_info ^(.+?\.php)(/.*)$; 
   
 to:
    location ~ ^\/nextcloud\/(?:index|remote|public|cron|core\/ajax\/update|status|ocs\/v[12]|updater\/.+|oc[ms]-provider\/.+|.+\/richdocumentscode\/proxy) /index.php$request_uri;
                                                                                                                                                                                                            
            fastcgi_split_path_info ^(.+?\.php)(/.*)$;

**Comment by dkahn93** [comment]
I used @alexbelgium 's Portainer Add-on for HASS, and deployed this stack:

`version: "2.2"
services:
  collabora:
    image: collabora/code
    container_name: collabora
    network_mode: bridge
    privileged: true
    ports:
      - 9980:9980`


I think it requires https, so I created a subdomain for collabora through the cloudflare add on.

In Nextcloud I went to administrative settings and then "Office", entered the collabora URL after checking the "use your own server". It should work after that.
